src/base.jl

```diff
@@ -143,65 +51,56 @@ function System(components, forecasts, basepower)
     return System(components, forecasts, basepower, PowerSystemInternal())
 end
 
-function System(sys::_System)
-    components = Dict{DataType, Vector{<:Component}}()
-    if !isnothing(sys.forecasts) && !isempty(sys.forecasts)
-        error("Constructing a System with an array of forecasts is not supported")
-    end
+"""System constructor when components are constructed externally."""
+function System(buses::Vector{Bus},
+                generators::Vector{<:Generator},
+                loads::Vector{<:ElectricLoad},
+                branches::Union{Nothing, Vector{<:Branch}},
+                storage::Union{Nothing, Vector{<:Storage}},
+                basepower::Float64,
+                forecasts::Union{Nothing, SystemForecasts},
+                services::Union{Nothing, Vector{ <: Service}},
+                annex::Union{Nothing,Dict}; kwargs...)
 
-    forecasts = SystemForecasts()
-    concrete_sys = System(components, forecasts, sys.basepower)
+    components = Dict{DataType, Vector{<:Component}}()
 
-    for field in (:buses, :loads)
-        for obj in getfield(sys, field)
-            add_component!(concrete_sys, obj)
-        end
+    if isnothing(forecasts)
+        forecasts = SystemForecasts()
     end
+    sys = System(components, forecasts, basepower)
 
-    for field in (:thermal, :renewable, :hydro)
-        generators = getfield(sys.generators, field)
-        if !isnothing(generators)
-            for gen in generators
-                add_component!(concrete_sys, gen)
-            end
-        end
+    arrays = [buses, generators, loads]
+    if !isnothing(branches)
+        push!(arrays, branches)
+    end
+    if !isnothing(storage)
+        push!(arrays, storage)
+    end
+    if !isnothing(services)
+        push!(arrays, services)
     end
 
-    for field in (:branches, :storage, :services)
-        objs = getfield(sys, field)
-        if !isnothing(objs)
-            for obj in objs
-                add_component!(concrete_sys, obj)
-            end
-        end
+    for component in Iterators.flatten(arrays)
+        add_component!(sys, component)
     end
 
-    load_zones = isnothing(sys.annex) ? nothing : get(sys.annex, :LoadZones, nothing)
+    load_zones = isnothing(annex) ? nothing : get(annex, :LoadZones, nothing)
     if !isnothing(load_zones)
         for lz in load_zones
-            add_component!(concrete_sys, lz)
+            add_component!(sys, lz)
         end
     end
 
-    for (key, value) in concrete_sys.components
+    for (key, value) in sys.components
         @debug "components: $(string(key)): count=$(string(length(value)))"
     end
 
-    return concrete_sys
-end
+    runchecks = get(kwargs, :runchecks, true)
+    if runchecks
```

Comment:
@KyraDiF Note this code path.  I modified it so that it would run in our new parsing paradigm.  These checks and modifications should probably happen in the validation code you are writing.  We will likely need some discussion about it.